syncer/syncer.go

```diff
@@ -451,6 +464,80 @@ func (s *Syncer) adjustWorkers() bool {
 	return shouldCreate
 }
 
+func (s *Syncer) handleSeenBlock(
+	ctx context.Context,
+	result *blockResult,
+) error {
+	// If the helper returns ErrOrphanHead
+	// for a block fetch, result.block will
+	// be nil.
+	if result.block == nil {
+		return nil
+	}
+
+	if err := s.seenSemaphore.Acquire(ctx, semaphoreWeight); err != nil {
```

Comment:
nit: should this be managed by the handler?

Reply:
I'm leaning towards yes!

Reply:
I don't have no strong opinion about this!